### ISSUE TITLE
Exception in Language Server thrown when adding a database

### ISSUE THREAD

**Issue by aeisenberg** [github/vscode-codeql#631]
Exceptions like this from the language server are being thrown when adding a database:

```
Oct. 21, 2020 8:04:02 A.M. org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleNotification
WARNING: Notification threw an exception: {
  "jsonrpc": "2.0",
  "method": "workspace/didChangeWorkspaceFolders",
  "params": {
    "event": {
      "added": [
        {
          "uri": "codeql-zip-archive://0-215/Users/andrew.eisenberg/Library/Application%20Support/Code/User/workspaceStorage/600c0c6f9b1cb74bf3beb8a937563b05/GitHub.vscode-codeql/esbena_bootstrap-pre-27047_javascript-6/esbena_bootstrap-pre-27047_204f9cf/src.zip/",
          "name": "[esbena_bootstrap-pre-27047_204f9cf source archive]"
        }
      ],
      "removed": []
    }
  }
}
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:152)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:220)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:187)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at com.semmle.queryserver.client.JsonRpcLauncher.lambda$startListening$0(JsonRpcLauncher.java:50)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1771)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:835)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 11 more
Caused by: com.semmle.util.exception.CatastrophicError: Cannot interpret URI as a path: codeql-zip-archive://0-215/Users/andrew.eisenberg/Library/Application%20Support/Code/User/workspaceStorage/600c0c6f9b1cb74bf3beb8a937563b05/GitHub.vscode-codeql/esbena_bootstrap-pre-27047_javascript-6/esbena_bootstrap-pre-27047_204f9cf/src.zip/
	at com.semmle.frontend.server.LanguageServerUtils.uriToPath(LanguageServerUtils.java:73)
	at com.semmle.frontend.server.QLLanguageServer$QLWorkspaceService.lambda$didChangeWorkspaceFolders$2(QLLanguageServer.java:521)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at com.semmle.frontend.server.QLLanguageServer$QLWorkspaceService.didChangeWorkspaceFolders(QLLanguageServer.java:522)
	... 16 more

```

The problem has to do with the language server not being able to handle the `codeql-zip-archive` schema.  Not sure if the fix is on the extension side or the server side.

**Comment by shad0wrunner** [comment]
Similar issue with my database.
I tried to use "Choose Database from Folder" option on my newly created database with JS-code
It imports fine but VSCode shows "Unable to resolve the workspace" error against the folder which was added by the extension.

Log file shows the following error:
[2020-10-27 18:45:08.987] [renderer1] [warning] Ignoring the error while validating workspace folder codeql-zip-archive://1-20/c%3A%5Cswd%5Cswd2%5Csrc.zip - Unable to resolve non-existing file 'codeql-zip-archive://1-20/c:\swd\swd2\src.zip'

Seems like the same issue with the schema.
